### PR TITLE
MCKIN-5637, MCKIN-6563 - Merge HTML XBlock student_view_data from Upstream

### PR DESCRIPTION
**Pull request by macornwell**
This pull request merges the changes found in upstream [PR #15905.](https://github.com/edx/edx-platform/pull/15905) In addition to the changes, other linked files such as openedx/core/djangoapps/content/block_structure/factory.py have been added.

This encompasses two changes:
1. Need to set settings.FEATURES['ENABLE_HTML_XBLOCK_STUDENT_VIEW_DATA'] to continue to see HTML `student_view_data`.
2. A bugfix for the Course Blocks API `student_view_data` parameter has been put in for testing when `settings.DEBUG = True`, because this change may affect the mobile apps using the Course Blocks API.
  Previously, if you put anything in that parameter, then all blocks with student_view_data would return theirs. With this fix enabled, the behavior is in line with [the documentation](http://edx.readthedocs.io/projects/edx-platform-api/en/latest/courses/blocks.html#query-parameters).

**JIRA tickets**: MCKIN-5637

**Testing instructions**:
The testing instructions are the same as [PR#15905](https://github.com/edx/edx-platform/pull/15905). Please see and use those.

**Reviewers**
- [ ] @pomegranited 
- [ ] TBD

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @macornwell! It looks like you're a member of a company that does contract work for edX. If you're doing this work as part of a paid contract with edX, you should talk to edX about who will review this pull request. If this work is not part of a paid contract with edX, then you should ensure that there is an OSPR issue to track this work in [JIRA](https://openedx.atlassian.net), so that we don't lose track of your pull request.

<a   href="https://openedx-webhooks.herokuapp.com/github/process_pr?repo=edx-solutions%2Fedx-platform&number=969"   rel="nofollow" >Create an OSPR issue for this pull request</a>.

File: lms/envs/common.py [hunk 1]
```diff
@@ -388,6 +388,39 @@
...
+
+    # Whether the bulk enrollment view is enabled.
+    'ENABLE_BULK_ENROLLMENT_VIEW': False,
+
+    # Whether course goals is enabled.
+    'ENABLE_COURSE_GOALS': True,
+
+    # Set to enable Enterprise integration
+    'ENABLE_ENTERPRISE_INTEGRATION': False,
+
```
Comment (bradenmacdonald) [inline lms/envs/common.py]: ^ everything above this line should be excluded from this PR - we only want the two new lines below

File: openedx/core/djangoapps/content/block_structure/block_structure.py [hunk 2]
```diff
@@ -0,0 +1,844 @@
+"""
```
Comment (bradenmacdonald) [inline openedx/core/djangoapps/content/block_structure/block_structure.py]: This is a whole new file - it should not be included.

File: lms/djangoapps/course_api/blocks/transformers/tests/test_student_view.py [hunk 3]
```diff
@@ -1,16 +1,17 @@
 """
 Tests for StudentViewTransformer.
 """
+import ddt
 
 # pylint: disable=protected-access
-
-from openedx.core.lib.block_structure.factory import BlockStructureFactory
+from openedx.core.djangoapps.content.block_structure.factory import BlockStructureFactory
```
Comment (bradenmacdonald) [inline lms/djangoapps/course_api/blocks/transformers/tests/test_student_view.py]: I think this should be using the old location, and you should not include the new `openedx/core/djangoapps/content/block_structure/factory.py` file.

Comment (macornwell) [reply lms/djangoapps/course_api/blocks/transformers/tests/test_student_view.py]: Didn't know there was an old one! Let me take care of that and remove all the added files.

File: common/lib/xmodule/xmodule/tests/test_html_module.py [hunk 4]
```diff
@@ -1,6 +1,12 @@
 import unittest
-import ddt
 from mock import Mock
+import ddt
+
+from django.test.utils import override_settings
+
+from opaque_keys.edx.locator import CourseLocator
+from xblock.field_data import DictFieldData
+from xblock.fields import ScopeIds
```
Comment (pomegranited) [inline common/lib/xmodule/xmodule/tests/test_html_module.py]: @macornwell  Please remove these 3 lines to fix the CircleCI tests:
* `CourseLocator` is not used, and was flagged by pylint
* `DictFieldData` is imported below
* `ScopeIds` is imported below

Comment (macornwell) [reply common/lib/xmodule/xmodule/tests/test_html_module.py]: Fixed them.

**Comment by bradenmacdonald** [comment]
We should wait for MCKIN-6563 (and the feature flag to be set as mentioned in the comments on that ticket) before merging this.

**Comment by bradenmacdonald** [comment]
@pomegranited Sounds like we're good to merge this as long as you can confirm it's bound to `DEBUG`, which seems to be the case.

**Comment by pomegranited** [comment]
Thanks @bradenmacdonald , yes, the bugfix [only happens when `settings.DEBUG == True`](https://github.com/edx-solutions/edx-platform/pull/969/files#diff-4122e6fe5f90cd6fa03fd4938044450dR323).  Merging now.

**Pull request merged**